### DISCUSSION TITLE
How to get entropy of saturated liquid

### DISCUSSION THREAD

**Discussion by ox-en-frog** [CoolProp/CoolProp#2264] [Q&A] [answered]
Good evening!

I am looking for an elegant and reliable way to get the entropy of the saturated liquid of a mixture. For trying, I use a CO2+N2 mixture (an idea using Python and Refprop is preferred).

AS = CP.AbstractState("REFPROP", "CO2&N2")
AS.set_mole_fractions([.947,.053])

Would be good enough to have the data between, say, -40°C and the critical point (so trouble with solid matter is avoided). Doesn't matter if molar or mass entropy is the result.  Basically, the p,T of the saturated liquid line would be sufficient, I could use these data for data extraction. Is there maybe a way to split up the data I receive from get_phase_envelope_data()?

Any idea would be very much appreciated!

Thank you very much!
Harald

**Comment by ibell** [comment, marked as answer]
There are two answers, that are complementary:
1. Do as you have done, call ``AS.update(CP.QT_INPUTS, 1, T)``, and then ``AS.p()`` after. You might want to also call ``AS.build_phase_envelope("")`` before to built the phase envelope to help the iterations succeed.
2. You can interpolate within the data obtained from ``AS.get_phase_envelope_data()``, and then you can pass that density and composition data back into ``AS.update_with_guesses()``, which should allow for close to 100% success rate of iterative calculations, except for very close to the critical point

**Reply by ox-en-frog** [reply]
Thank you, Ian, very nice of you answering so quickly and precisely. Great program, thank you very much!
